sheetstorm/management/commands/generate_test_data.py

```diff
@@ -61,6 +75,65 @@ def _set_user_factory_parameters(user_type):
             "is_superuser": user_type == superuser_user_type,
         }
 
+    def execute_creating_project(self, options):
+        projects_to_create = self._set_number_of_projects_to_create(options)
+
+        for (project_type, number_of_projects) in projects_to_create.items():
+            self.create_project(project_type, number_of_projects)
+
+    def _set_number_of_projects_to_create(self, options):
+        return {
+            ProjectType.SUSPENDED.name: self._compute_number_of_suspended_projects_to_create(options),
+            ProjectType.ACTIVE.name: self._compute_number_of_active_projects_to_create(options),
+            ProjectType.COMPLETED.name: self._compute_number_of_completed_projects_to_create(options),
+        }
+
+    @staticmethod
+    def _compute_number_of_suspended_projects_to_create(options):
+        return (
+            options[ProjectType.SUSPENDED.name] - Project.objects.filter_suspended().count()
+            if options[ProjectType.SUSPENDED.name] is not None
+            else 0
+        )
+
+    @staticmethod
+    def _compute_number_of_active_projects_to_create(options):
+        return (
+            options[ProjectType.ACTIVE.name] - Project.objects.filter_active().count()
+            if options[ProjectType.ACTIVE.name] is not None
+            else 0
+        )
+
+    @staticmethod
+    def _compute_number_of_completed_projects_to_create(options):
+        return (
+            options[ProjectType.COMPLETED.name] - Project.objects.filter_completed().count()
+            if options[ProjectType.COMPLETED.name] is not None
+            else 0
+        )
+
+    def create_project(self, project_type, number_of_projects_to_create):
+        factory_parameters = self._set_project_factory_parameters(project_type)
+
+        for project_number in range(number_of_projects_to_create):
+            logging.info(f"{number_of_projects_to_create - project_number} {project_type} project(s) left to create")
+            ProjectFactory(**factory_parameters)
+
+    def _set_project_factory_parameters(self, project_type):
+        return {
+            "start_date": self._create_start_date(),
+            "suspended": project_type == ProjectType.SUSPENDED.name,
+            "stop_date": self._create_stop_date() if project_type == ProjectType.COMPLETED.name else None,
+        }
+
+    @staticmethod
+    def _create_start_date(time_delta=relativedelta(months=1, day=1)):
```

Comment:
I think that `time_delta` should be class variable and you should it here. Something like: `PROJECT_START_DATE_TIME_DELTA = relativedelta(months=1, day=1)` and then:
```python
def _create_start_date(time_delta=PROJECT_START_DATE_TIME_DELTA)
```

Reply:
Done